src/core_plugins/kibana/public/management/sections/settings/advanced_import.js

```diff
@@ -0,0 +1,40 @@
+import { uiModules } from 'ui/modules';
+import advancedImportTemplate from 'plugins/kibana/management/sections/settings/advanced_import.html';
+import { AdvancedImportState } from 'plugins/kibana/management/sections/settings/advanced_import_state.js';
+
+uiModules.get('apps/management')
+  .directive('advancedImport', function (config, Private) {
+    const advancedImportState = Private(AdvancedImportState);
+
+    const abandonSetting = setting => {
+      config.set(setting.name, setting.oldValue);
+      advancedImportState.markAsAbandoned(setting);
+    };
+
+    const importSetting = setting => {
+      config.set(setting.name, setting.value);
+      advancedImportState.markAsImported(setting);
+    };
+
+    return {
+      restrict: 'E',
+      replace: true,
+      template: advancedImportTemplate,
+      controllerAs: 'advancedImport',
+      bindToController: true,
+      controller: class AdvancedImportController {
+        constructor() {
+          this.importState = advancedImportState;
```

Comment:
I'd prefer to see you use `this.importState` after this point. That way the alias is consistent here and in the template.

You can also move `abandonSetting` and `importSetting` into the constructor, or better, since you've got a class here, make them methods on the class object.